contributors/design-proposals/pod-resource-management.md

```diff
@@ -1,114 +1,419 @@
+meaning than `500m` of cpu for a Burstable pod in the current hierarchy.
+For this reason, we will always set `cpu.shares` for the QoS level sandboxes
+by default as part of roll-out for this feature.
+
+### QoS level memory allocation
+
+By default, no memory limits are applied to the BestEffort
+and Burstable QoS level cgroups unless a `--qos-reserve-requests` value
+is specified for memory.
+
+The heuristic that is applied is as follows for each QoS level sandbox:
+
+```
+ROOT/Burstable/memory.limit_in_bytes = 
+    Node.Allocatable - {(summation of memory requests of `Guaranteed` pods)*(reservePercent / 100)}
+ROOT/BestEffort/memory.limit_in_bytes = 
+    Node.Allocatable - {(summation of memory requests of all `Guaranteed` and `Burstable` pods)*(reservePercent / 100)}
+```
+
+A value of `--experimental-qos-reserve-requests=memory=100%` will cause the
+`kubelet` to adjust the Burstable and BestEffort cgroups from consuming memory
+that was requested by a higher QoS class. This increases the risk
+of inducing OOM on BestEffort workloads in favor of increasing memory
+resource guarantees for Burstable and Guaranteed workloads.  A value of 
+`--experimental-qos-reserve-requests=memory=0%` will allow a Burstable
+and BestEffort QoS sandbox to consume up to the full allocatable amount if
+available in favor of increasing memory resource guarantees for Guaranteed
+workloads.
+
+Since memory is an incompressible resource, it is possible that a QoS
+level cgroup sandbox may not be able to reduce memory usage below the
+value specified in the heuristic during pod admission and pod termination.
+As a result, the `kubelet` runs a periodic thread to attempt to converge
+to this desired state from the above heuristic.  If unreclaimable memory
+usage has exceeded the desired limit for the sandbox, the `kubelet` will
+attempt to set the effective limit near the current usage to put pressure
+on the QoS cgroup sandbox and prevent further consumption.
+
+Adjusting the memory limits for the QoS level cgroup sandbox carries
+greater risk given the incompressible nature of memory.  As a result,
+we are not enabling this function by default, but would like operators
+that want to value resource priority over resource utilization to gather
+real-world feedback on its utility.  
+
+## Memory backed volumes
+
+The pod level cgroup ensures that any writes to a memory backed volume
+are correctly charged to the pod sandbox even when a container process
+in the pod restarts.
+
+All memory backed volumes are removed when a pod reaches a terminal state.
+
+The `kubelet` verifies that a pod's cgroup is deleted from the
+host before deleting a pod from the API server as part of the 
+
+This ensures resource consumption associated with those volumes are not
+incorrectly transferred back to the node and induces unintentional overcommit.
+
+## Log basic cgroup management
+
+The `kubelet` will log and collect metrics associated with cgroup manipulation.
```

Comment:
this is a TODO, i am hoping between @sjenning and me, we can bang a PR out by monday.

we basically will need an instrumented CgroupManager